feat: allow Playwright to run on Android with opt-in env flag

Jobians opened the pull request:
### Summary

This change allows advanced users to run Playwright on Android (e.g. via Termux) by setting an environment variable `PLAYWRIGHT_ALLOW_ANDROID=1`.

By default, Android remains unsupported. This opt-in flag is useful for cases where users provide their own Chromium and are aware of the limitations.

### Motivation

Currently, Playwright throws an "Unsupported platform: android" error even if Chromium works and headless scripts function correctly in Termux.

This patch:
- Keeps current platform safety behavior
- Adds a safe override for power users
- Helps open up use cases for scripting, scraping, and automation on Android

### Notes

- No browser binaries are bundled or downloaded
- Chromium must be installed manually
- Tested successfully with `playwright-core` + Termux + Chromium + pnpm + tsx

Jobians committed:
feat: allow Playwright to run on Android with opt-in env flag

### Summary

This change allows advanced users to run Playwright on Android (e.g. via Termux) by setting an environment variable `PLAYWRIGHT_ALLOW_ANDROID=1`.

By default, Android remains unsupported. This opt-in flag is useful for cases where users provide their own Chromium and are aware of the limitations.

### Motivation

Currently, Playwright throws an "Unsupported platform: android" error even if Chromium works and headless scripts function correctly in Termux.

This patch:
- Keeps current platform safety behavior
- Adds a safe override for power users
- Helps open up use cases for scripting, scraping, and automation on Android

### Notes

- No browser binaries are bundled or downloaded
- Chromium must be installed manually
- Tested successfully with `playwright-core` + Termux + Chromium + pnpm + tsx

Signed-off-by: JOBIANS TECHIE <88005779+Jobians@users.noreply.github.com>

yury-s commented on packages/playwright-core/src/server/registry/index.ts:
```diff
@@ -406,8 +406,11 @@ export const registryDirectory = (() => {
     else if (process.platform === 'win32')
       cacheDirectory = process.env.LOCALAPPDATA || path.join(os.homedir(), 'AppData', 'Local');
-    else
+    else if (process.platform === 'android' && process.env.PLAYWRIGHT_ALLOW_ANDROID === '1') {
```
This code is only used when installing bundled browsers and we don't provide those for android anyway, so the user shouldn't need to run `npx playwright install` on android.

Can you start with filing a bug/feature request that would describe in more details the user workflow that you want to support and at what step it currently fails?

Jobians replied:
Just to clarify — the error I’m addressing **does not occur during `npx playwright install`**, but rather **at runtime**, when I run my own script using `playwright-core`:

```bash
npx tsx src/scraper.ts
```

I'm using `playwright-core` without bundled browsers and manually providing the browser path:

```ts
import { chromium } from 'playwright-core';

const browser = await chromium.launch({
  executablePath: '/data/data/com.termux/files/usr/bin/chromium-browser',
  headless: true,
});
```

Even though I'm supplying `executablePath`, Playwright still tries to access the browser registry and throws this error:

```
Error: Unsupported platform: android
```

That error originates from this block in `registry/index.ts`:

```ts
else {
  throw new Error('Unsupported platform: ' + process.platform);
}
```

To prevent that, I added this platform check:

```ts
else if (process.platform === 'android' && process.env.PLAYWRIGHT_ALLOW_ANDROID === '1') {
  cacheDirectory = path.join(os.homedir(), '.cache');
}
```

This change ensures Playwright doesn't fail on Android when:
- The user explicitly sets `PLAYWRIGHT_ALLOW_ANDROID=1`
- A valid `executablePath` is already provided
- No browser install is required

The registry path is still accessed even during manual browser launches, so this fix is needed to avoid blocking runtime usage on Android in valid cases.

Jobians replied:
Setting `PLAYWRIGHT_BROWSERS_PATH=0` and using `executablePath` to point to the Termux-installed Chromium lets Playwright work on Termux despite the platform not being officially supported.

Thanks

Jobians committed:
feat(core): add Android platform support with PLAYWRIGHT_ALLOW_ANDROID

## 🐧 Add support for Android platform (e.g. Termux)

Some custom Node.js builds (e.g. in Termux) return `process.platform === 'android'`.  
Currently, Playwright throws an "Unsupported platform" error in these environments — even when the user provides a valid browser path and uses `playwright-core` manually.

---

### ✅ What this change does

- Adds a platform check for `process.platform === 'android'`.
- Requires the user to set `PLAYWRIGHT_ALLOW_ANDROID=1` to proceed, similar to other unsupported platforms.
- Sets the browser registry path to `~/.cache`, which is appropriate for Termux and other Android-based environments.

---

### 📌 Why this matters

This enables valid use cases where:

- Users are running Node.js in Termux or Android environments.
- They manually install and launch Chromium using `playwright-core`.
- Automatic browser installation is not required, but Playwright currently errors at runtime anyway.

---

### 💡 Example user workflow

```ts
import { chromium } from 'playwright-core';

const browser = await chromium.launch({
  executablePath: '/data/data/com.termux/files/usr/bin/chromium-browser',
  headless: true
});
```

This works with a valid Chromium binary, but fails unless PLAYWRIGHT_ALLOW_ANDROID=1 is set.
This change enables that scenario safely and intentionally.

Signed-off-by: JOBIANS TECHIE <88005779+Jobians@users.noreply.github.com>

yury-s commented:
Closing as Android is not a [supported platform](https://playwright.dev/docs/intro#system-requirements) and we don't want to mislead the users.
